stripe/resource_stripe_meter.go

```diff
@@ -0,0 +1,299 @@
+			}
+		}
+	}
+
+	if valueSettings, set := d.GetOk("value_settings"); set {
+		params.ValueSettings = &stripe.BillingMeterValueSettingsParams{}
+		valueSettingsMap := ToMap(valueSettings)
+		for k, v := range valueSettingsMap {
+			switch {
+			case k == "event_payload_key" && ToString(v) != "":
+				params.ValueSettings.EventPayloadKey = stripe.String(ToString(v))
+			}
+		}
+	}
+
+	err = retryWithBackOff(func() error {
+		meter, err = c.BillingMeters.New(params)
+		return err
+	})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(meter.ID)
+	return resourceStripeMeterRead(ctx, d, m)
+}
+
+func resourceStripeMeterUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*client.API)
+	var err error
+
+	params := &stripe.BillingMeterParams{}
+
+	if d.HasChange("display_name") {
+		params.DisplayName = stripe.String(ExtractString(d, "display_name"))
+	}
+
+	if d.HasChange("event_name") {
+		params.EventName = stripe.String(ExtractString(d, "event_name"))
+	}
+
+	if d.HasChange("event_time_window") {
+		params.EventTimeWindow = stripe.String(ExtractString(d, "event_time_window"))
+	}
+
+	if d.HasChange("default_aggregation") {
+		params.DefaultAggregation = &stripe.BillingMeterDefaultAggregationParams{}
+		defaultAggregationMap := ToMap(ExtractMap(d, "default_aggregation"))
+		for k, v := range defaultAggregationMap {
+			switch {
+			case k == "formula" && ToString(v) != "":
+				params.DefaultAggregation.Formula = stripe.String(ToString(v))
+			}
+		}
+	}
+
+	if d.HasChange("customer_mapping") {
+		params.CustomerMapping = &stripe.BillingMeterCustomerMappingParams{}
+		customerMappingMap := ToMap(ExtractMap(d, "customer_mapping"))
+		for k, v := range customerMappingMap {
+			switch {
+			case k == "event_payload_key" && ToString(v) != "":
+				params.CustomerMapping.EventPayloadKey = stripe.String(ToString(v))
+			case k == "type" && ToString(v) != "":
+				params.CustomerMapping.Type = stripe.String(ToString(v))
+			}
+		}
+	}
+
+	if d.HasChange("value_settings") {
+		params.ValueSettings = &stripe.BillingMeterValueSettingsParams{}
+		valueSettingsMap := ToMap(ExtractMap(d, "value_settings"))
+		for k, v := range valueSettingsMap {
+			switch {
+			case k == "event_payload_key" && ToString(v) != "":
+				params.ValueSettings.EventPayloadKey = stripe.String(ToString(v))
+			}
+		}
+	}
+
+	err = retryWithBackOff(func() error {
+		_, err = c.BillingMeters.Update(d.Id(), params)
+		return err
+	})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return resourceStripeMeterRead(ctx, d, m)
+}
+
+func resourceStripeMeterDelete(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*client.API)
+	var err error
+
+	params := stripe.BillingMeterDeactivateParams{}
+
+	err = retryWithBackOff(func() error {
+		_, err = c.BillingMeters.Deactivate(d.Id(), &params)
+		return err
+	})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
```

Comment:
There should be a log as it is in other resources when the delete operation is not a delete but deactivate.

Reply:
done